app/src/main/java/org/eyeseetea/malariacare/presenter/ReviewPresenter.java

```diff
@@ -0,0 +1,122 @@
+                        value.getOption().getInternationalizedCode());
+            }
+            if (colorIterator.hasNext()) {
+                preparedValue.setBackgroundColor(colorIterator.next());
+            }
+            preparedValues.add(preparedValue);
+        }
+        return preparedValues;
+    }
+
+    private List<String> createBackgroundColorList(
+            List<org.eyeseetea.malariacare.data.database.model.Value> values) {
+        List<String> colorsList = new ArrayList<>();
+        for (org.eyeseetea.malariacare.data.database.model.Value value : values) {
+            if (value.getOption() != null && value.getOption().getBackground_colour() != null) {
+                String color = "#" + value.getOption().getBackground_colour();
+                if (!colorsList.contains(color)) {
+                    colorsList.add(color);
+                }
+            }
+        }
+        //Hardcoded colors for a colorList without colors.
+        if (colorsList.size() == 0) {
+            colorsList.add("#4d3a4b");
+        }
+        if (colorsList.size() == 1 && values.size() > 1) {
+            colorsList.add("#9c7f9b");
+        }
+        return colorsList;
+    }
+
+    private List<org.eyeseetea.malariacare.data.database.model.Value> orderValues(
+            List<org.eyeseetea.malariacare.data.database.model.Value> values) {
+        return ReviewFragmentStrategy.orderValues(values);
+    }
+
+    private List<org.eyeseetea.malariacare.data.database.model.Value> getReviewValues() {
+        List<org.eyeseetea.malariacare.data.database.model.Value> reviewValues = new ArrayList<>();
+        Survey survey = Session.getMalariaSurvey();
+        List<org.eyeseetea.malariacare.data.database.model.Value> allValues =
+                survey.getValuesFromDB();
+        for (org.eyeseetea.malariacare.data.database.model.Value value : allValues) {
+            boolean isReviewValue = true;
+            if (value.getQuestion() == null) {
+                continue;
+            }
+            for (QuestionRelation questionRelation : value.getQuestion().getQuestionRelations()) {
+                if (questionRelation.isACounter() || questionRelation.isAReminder()
+                        || questionRelation.isAWarning() || questionRelation.isAMatch()) {
+                    isReviewValue = false;
+                }
+            }
+            int output = value.getQuestion().getOutput();
+            if (output == Constants.HIDDEN
+                    || output == Constants.DYNAMIC_STOCK_IMAGE_RADIO_BUTTON) {
+                isReviewValue = false;
+            }
+            if (isReviewValue) {
+                if (value.getQuestion() != null) {
+                    reviewValues.add(value);
+                }
+            }
+        }
+        return reviewValues;
+    }
+
+    @Override
+    public void onClickOnValue(String UId) {
+        DashboardActivity.dashboardActivity.hideReview(UId);
```

Comment:
Presenter should not invoke directly to presentation object. Only can invoke methods of ReviewView to communicate with presentation layer. Can you add navigateToQuestion(String uid) to ReviewView?